### ISSUE TITLE
Not able to use pinia: No matching export in "node_modules/pinia/node_modules/vue-demi/lib/index.mjs" for import "hasInjectionContext"

### ISSUE THREAD

**Issue by shree-c** [posva/pinia#2210]
### Reproduction

https://github.com/shree-c/mock_two_test_pinia

### Steps to reproduce the bug

![Screenshot from 2023-05-17 15-26-02](https://github.com/vuejs/pinia/assets/58801143/09d13768-8c70-49aa-a4a2-cd8da4a9056c)
![Screenshot from 2023-05-17 15-29-38](https://github.com/vuejs/pinia/assets/58801143/81e3b3af-210e-4786-8d97-7d6f394aa829)

### Expected behavior

I am doing basic activity such as creating a store and importing it into a component.

### Actual behavior

Vite not able to build both for dev and prod.
I am getting the following error on vite terminal:
`No matching export in "node_modules/pinia/node_modules/vue-demi/lib/index.mjs" for import "hasInjectionContext"`

**Comment by posva** [comment]
You need to update vue to 3.3. I should have set the peer dep...

**Comment by shunyue1320** [comment]
"hasInjectionContext" is not exported by "node_modules/.pnpm/pinia@2.1.1_typescript@5.0.4+vue@3.3.3/node_modules/vue-demi/lib/index.mjs", imported by "node_modules/.pnpm/pinia@2.1.1_typescript@5.0.4+vue@3.3.3/node_modules/pinia/dist/pinia.mjs".

![image](https://github.com/vuejs/pinia/assets/47295879/bd76c0e8-96ae-47dd-a291-fb373c17af51)

**Comment by temofart** [comment]
@shree-c @shunyue1320 hey guys, any updated? i have the same issue while migrating from vue-cli to vite

**Comment by lautaandreani** [comment]
For me, work update to vue 3.3.4
```
"pinia": "^2.1.3",
"vue": "^3.3.4",
```

**Comment by MickeyMouse623** [comment]
<img width="897" alt="image" src="https://github.com/vuejs/pinia/assets/18524460/685f51a9-a949-4b37-bee8-9b74bd423d36">
还是不行

**Comment by MickeyMouse623** [comment]
> 

<img width="775" alt="image" src="https://github.com/vuejs/pinia/assets/18524460/e0cbcd11-634c-412e-b33c-53d898e68ed9">
这个版本配置正常运行

**Comment by BeidouGoan** [comment]
#

> > 
> 
> <img alt="image" width="775" src="https://user-images.githubusercontent.com/18524460/244682986-e0cbcd11-634c-412e-b33c-53d898e68ed9.png"> 这个版本配置正常运行
只能说与vue@3.3+pinia@2.1无缘

**Comment by aghArdeshir** [comment]
hey @posva sorry for digging up this old issue. I also face this problem when trying to link pinia locally.

In my case I am using vue@2.7.14; what is the solution in my case?

**Comment by outNapGnaw** [comment]
![image](https://github.com/vuejs/pinia/assets/13870052/5290a6b6-122d-4a17-a57b-071d7821dabd)
依然存在报错

**Comment by MikeMitterer** [comment]
Argg.... had to downgrade to 2.0.35 because of this bug. @posva There are a lot of 2.7 installations in the field. Upgrading to vue 3.x is very often not possible. So to others with the same problem: Vue 2.7 only works with Pinia 2.0.35 (and below???, not tested)

**Comment by temofart** [comment]
I just upgraded vue and pinia both to the latest versions and everything is ok

**Comment by posva** [comment]
For vue 2 update vue-demi so it has the correct exports

**Comment by 869288142** [comment]
> > > 
> > 
> > 
> > <img alt="image" width="775" src="https://user-images.githubusercontent.com/18524460/244682986-e0cbcd11-634c-412e-b33c-53d898e68ed9.png"> 这个版本配置正常运行
> > 只能说与vue@3.3+pinia@2.1无缘

if you use uni-app, it's inner vue version is 3.2.47. we must be use pinia 2.0.x

**Comment by chengazhen** [comment]
uniapp 使用pinna必须使用固定的版本

**Comment by JasonCareter** [comment]
少了一行导出而已，加上就完了，yarn install后执行一行命令把代码添加进index.mjs就行了
echo 'export function hasInjectionContext() {return !!Vue.getCurrentInstance()}' >> ./node_modules/vue-demi/lib/index.mjs

**Comment by jaythomas** [comment]
The fix for a fresh vite + vue 2.7 app was to downgrade pinia from 2.1.7 to 2.0.x. Seems there was some regression in the dependency chain. Hope this helps the next person that sees this GitHub issue in the search results trying to troubleshoot this problem and hopefully this issue will be resolved in a future version of pinia.

**Comment by khamaileon** [comment]
Fix for vue-demi build issue with PNPM in GitHub Actions

If you're getting this error during pnpm build:

`export 'hasInjectionContext' was not found in 'vue-demi'`

And PNPM shows a warning about ignored build scripts, the issue is that PNPM is blocking vue-demi's build script, preventing it from properly detecting Vue 2 or Vue 3.

Solution:
Add this step before your build in GitHub Actions:

```
- name: Approve vue-demi build
  run: pnpm approve-builds
```

This will allow vue-demi to execute its required setup and fix the issue. 🚀
